integration-tests/metrics-init.patch

```diff
@@ -0,0 +1,99 @@
+         try {
+             return TimeUnit.valueOf(String.valueOf(value).toUpperCase(Locale.US));
+--- a/metrics-servlets/src/main/java/io/dropwizard/metrics5/servlets/MetricsServlet.java
++++ b/metrics-servlets/src/main/java/io/dropwizard/metrics5/servlets/MetricsServlet.java
+@@ -188,6 +188,9 @@ public class MetricsServlet extends HttpServlet {
+         return mapper.writer();
+     }
+ 
++    // Here `value` may be `null`, while `TimeUnit#valueOf` requires a non-`null` argument.
++    // XXX: Investigate better nullness handling by `IdentityConversion`.
++    @SuppressWarnings("IdentityConversion")
+     protected TimeUnit parseTimeUnit(String value, TimeUnit defaultValue) {
+         try {
+             return TimeUnit.valueOf(String.valueOf(value).toUpperCase(Locale.US));
+--- a/pom.xml
++++ b/pom.xml
+@@ -136,6 +136,27 @@
+         </repository>
+     </distributionManagement>
+ 
++    <dependencyManagement>
++        <dependencies>
++            <dependency>
++                <groupId>com.google.errorprone</groupId>
++                <artifactId>error_prone_annotations</artifactId>
++                <version>${error-prone.version}</version>
++            </dependency>
++            <dependency>
++                <groupId>com.google.guava</groupId>
++                <artifactId>guava</artifactId>
++                <version>33.3.1-jre<!-- Renovate: com.google.guava:guava-bom --></version>
++            </dependency>
++        </dependencies>
++    </dependencyManagement>
++    <dependencies>
++        <dependency>
++            <groupId>com.google.guava</groupId>
++            <artifactId>guava</artifactId>
++        </dependency>
++    </dependencies>
++
+     <profiles>
+         <profile>
+             <id>jdk8</id>
+@@ -218,7 +239,7 @@
+                             <compilerArgs>
+                                 <arg>-Xlint:all</arg>
+                                 <arg>-XDcompilePolicy=simple</arg>
+-                                <arg>-Xplugin:ErrorProne -XepExcludedPaths:.*/target/generated-sources/.*</arg>
++                                <arg>-Xplugin:ErrorProne ${error-prone.configuration-args}</arg>
+                                 <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED</arg>
+                                 <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
+                                 <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
+@@ -229,12 +250,24 @@
+                                 <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
+                                 <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
+                                 <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
++                                <arg>-Xmaxwarns</arg>
++                                <arg>1000000</arg>
```

Comment:
I added this for consistency with the Checkstyle setup. The fact that _right now_ it doesn't have impact is a function of our checks and the code under test.

(The alternative is to drop `-Xmaxwarns` from both integration tests. This reduces patch maintenance at the cost of a potentially incomplete list of warnings. I guess it's fine to consistently apply the flag, for now.)